light-node/src/commands/start.rs

```diff
@@ -0,0 +1,180 @@
+//! `start` subcommand - example of how to write a subcommand
+
+/// App-local prelude includes `app_reader()`/`app_writer()`/`app_config()`
+/// accessors along with logging macros. Customize as you see fit.
+use crate::prelude::*;
+
+use core::future::Future;
+use tendermint::hash;
+use tendermint::lite;
+use tendermint::lite::{Error, Header, Requester, SignedHeader, Store, TrustedState};
+use tendermint::rpc;
+use tendermint::{block::Height, Hash};
+use tokio::runtime::Builder;
+
+use tendermint::lite::ValidatorSet as _;
+
+use crate::config::LightNodeConfig;
+use crate::requester::RPCRequester;
+use crate::state::State;
+use crate::store::MemStore;
+use crate::threshold::TrustThresholdOneThird;
+
+use abscissa_core::{config, Command, FrameworkError, Options, Runnable};
+use std::time::{Duration, SystemTime};
+
+/// `start` subcommand
+///
+/// The `Options` proc macro generates an option parser based on the struct
+/// definition, and is defined in the `gumdrop` crate. See their documentation
+/// for a more comprehensive example:
+///
+/// <https://docs.rs/gumdrop/>
+#[derive(Command, Debug, Options)]
+pub struct StartCmd {
+    /// RPC address to request headers and validators from.
+    #[options(free)]
+    rpc_addr: String,
+}
+
+// TODO: this should also somehow be configurable ...
+// we can't simply add this as a field in the config because this either would
+// be a trait (`TrustThreshold`) or immediately and impl thereof (`TrustThresholdOneThird`).
```

Comment:
What's the best UX for this? The trust threshold seems most naturally to be a fraction, like `1/3`, `2/5`, etc. So we could just provide an implementation of TrustThreshold that takes a numerator and denominator and applies it accordingly. We can keep TrustThresholdOneThird as a thing but it would just be TrustThresholdFraction::new(1, 3)

Reply:
>  So we could just provide an implementation of TrustThreshold that takes a numerator and denominator and applies it accordingly.

Yeah, I thought about this too. I think that would be the easiest and general enough!

Reply:
Refactored here: https://github.com/interchainio/tendermint-rs/pull/132/files#diff-b1b6c7ff77dbf13838f76b8403715185R88-R119